lib/api/model/model.dart

```diff
@@ -604,6 +613,33 @@ extension type const TopicName(String _value) {
   /// using [canonicalize].
   bool isSameAs(TopicName other) => canonicalize() == other.canonicalize();
 
+  /// Convert this topic to match how it would appear on a message object from
+  /// the server, assuming the topic is originally for a send-message request.
+  ///
+  /// For a client that does not support empty topics,
+  /// a modern server (FL>=334) would convert "(no topic)" and empty topics to
+  /// `store.realmEmptyTopicDisplayName`.
+  ///
+  /// See also: https://zulip.com/api/send-message#parameter-topic
+  TopicName interpretAsServer({
+    required int zulipFeatureLevel,
+    required String? realmEmptyTopicDisplayName,
+  }) {
+    assert(_value.trim() == _value);
```

Comment:
Good to have the assert, but let's mention the restriction on when this can be called. I guess the public-interface way of saying that is: it should only be called if the instance was constructed from a string without leading/trailing whitespace, right? (or "from a trimmed string" or similar)